browser/help/topics/v4-hts.md

```diff
@@ -66,11 +66,336 @@ The above example will retrieve the entire frequency struct for each variant. To
+- `region_flags`: The v4 exomes Hail Table has the following fields that are not present in the struct on the v4 genomes Hail Table:
+  - `fail_interval_qc`
+  - `outside_ukb_capture_region`
+  - `outside_broad_capture_region`
+- `allele_info`: The v4 exomes Hail Table has an additional field in this struct, 'has_star', that is not present in the struct on the v4 genomes Hail Table
+- `info`: Sibling singletons were used to train the variant QC models for the v4 exomes but were not used in the v4 genomes variant QC
+
+#### gnomAD v4.1 joint frequency Hail Table annotations
+
+The v4 genomes Hail Table annotation schema is the same as the exomes schema, with only a few minor differences:
+
+Global fields
+
+- `genomes_freq_meta`: Allele frequency metadata for the gnomAD genomes. An ordered list containing the frequency aggregation group for each element of the ‘freq’ array row annotation.
+- `genomes_faf_meta`: Filtering allele frequency metadata for the gnomAD genomes. An ordered list containing the frequency aggregation group for each element of the ‘faf’ array row annotation.
+- `exomes_freq_meta`: Allele frequency metadata for the gnomAD exomes. An ordered list containing the frequency aggregation group for each element of the ‘freq’ array row annotation.
+- `exomes_faf_meta`: Filtering allele frequency metadata for the gnomAD exomes. An ordered list containing the frequency aggregation group for each element of the ‘faf’ array row annotation.
+- `joint_freq_meta`: Joint allele frequency across the exomes and genomes metadata. An ordered list containing the frequency aggregation group for each element of the ‘joint_freq’ array row annotation.
+- `joint_freq_index_dict`: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the ‘joint_freq’ array row annotation.
+- `joint_faf_meta`: Joint filtering allele frequency across the exomes and genomes metadata. An ordered list containing the frequency aggregation group for each element of the ‘joint_faf’ array row annotation. joint_faf_index_dict: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the filtering allele frequency (‘joint_faf’) row annotation.
+- `joint_faf_index_dict`: Dictionary keyed by specified label grouping combinations (group: adj/raw, gen_anc: gnomAD inferred genetic ancestry group, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the filtering allele frequency (‘joint_faf’) row annotation.
+- `joint_sample_count`: Number of gnomAD exomes + number of gnomAD genomes.
+
+Row fields
+
```

Comment:
Also for Row fields, main change seems to be "region_flags" and "histograms".
```
Row fields:
    'locus': locus<GRCh38> 
    'alleles': array<str> 
    'region_flags': struct {
        fail_interval_qc: bool, 
        outside_ukb_capture_region: bool, 
        outside_broad_capture_region: bool, 
        outside_ukb_calling_region: bool, 
        outside_broad_calling_region: bool, 
        not_called_in_exomes: bool, 
        not_called_in_genomes: bool
    } 
    'genomes': struct {
        freq: array<struct {
            AC: int32, 
            AF: float64, 
            AN: int32, 
            homozygote_count: int32
        }>, 
        faf: array<struct {
            faf95: float64, 
            faf99: float64
        }>, 
        grpmax: struct {
            AC: int32, 
            AF: float64, 
            AN: int32, 
            homozygote_count: int32, 
            gen_anc: str
        }, 
        fafmax: struct {
            faf95_max: float64, 
            faf95_max_gen_anc: str, 
            faf99_max: float64, 
            faf99_max_gen_anc: str
        }, 
        histograms: struct {
            qual_hists: struct {
                gq_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                gq_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                ab_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }, 
            raw_qual_hists: struct {
                gq_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                gq_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                ab_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }, 
            age_hists: struct {
                age_hist_het: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                age_hist_hom: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }
        }
    } 
    'exomes': struct {
        freq: array<struct {
            AC: int32, 
            AF: float64, 
            AN: int32, 
            homozygote_count: int32
        }>, 
        faf: array<struct {
            faf95: float64, 
            faf99: float64
        }>, 
        grpmax: struct {
            AC: int32, 
            AF: float64, 
            AN: int32, 
            homozygote_count: int64, 
            gen_anc: str
        }, 
        fafmax: struct {
            faf95_max: float64, 
            faf95_max_gen_anc: str, 
            faf99_max: float64, 
            faf99_max_gen_anc: str
        }, 
        histograms: struct {
            qual_hists: struct {
                gq_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                gq_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                ab_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }, 
            raw_qual_hists: struct {
                gq_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                gq_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                ab_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }, 
            age_hists: struct {
                age_hist_het: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                age_hist_hom: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }
        }
    } 
    'joint': struct {
        freq: array<struct {
            AC: int32, 
            AF: float64, 
            AN: int32, 
            homozygote_count: int32
        }>, 
        faf: array<struct {
            faf95: float64, 
            faf99: float64
        }>, 
        fafmax: struct {
            faf95_max: float64, 
            faf95_max_gen_anc: str, 
            faf99_max: float64, 
            faf99_max_gen_anc: str, 
            joint_fafmax_data_type: str
        }, 
        grpmax: struct {
            AC: int32, 
            AF: float64, 
            AN: int32, 
            homozygote_count: int32, 
            gen_anc: str
        }, 
        histograms: struct {
            qual_hists: struct {
                gq_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                gq_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                ab_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }, 
            raw_qual_hists: struct {
                gq_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_all: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                gq_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                dp_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                ab_hist_alt: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }, 
            age_hists: struct {
                age_hist_het: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }, 
                age_hist_hom: struct {
                    bin_edges: array<float64>, 
                    bin_freq: array<int64>, 
                    n_smaller: int64, 
                    n_larger: int64
                }
            }
        }, 
        metric_data_type: str
    } 
    'freq_comparison_stats': struct {
        contingency_table_test: array<struct {
            odds_ratio: float64, 
            p_value: float64
        }>, 
        cochran_mantel_haenszel_test: struct {
            odds_ratio: float64, 
            p_value: float64
        }
    } 
```

Reply:
the exomes frequencies are also no longer stratified by subset (the exome grpmax field was stratified in a previous schema). just confirming this was intentional @jkgoodrich

Reply:
When I looked at the exomes release HT, `grpmax` and `fafmax` are still nested struct of `gnomad` and `non_ukb`, I think like last time, we only used these two from `gnomad` to get joint FAF. 
I don't see `freq` was stratified by subsets, in v4.0 release HT and help page.

Reply:
sounds good, it makes sense to me that you'd use exome-wide frequencies to create the joint freqs.

also, `freq` doesn't need to be stratified because the subset information is in the `freq_meta` stratification (e.g, `{'downsampling': '200000', 'gen_anc': 'global', 'group': 'adj', 'subset': 'non_ukb'}`)

Reply:
I'm not 100% sure now; that might have been me checking the schema of the v4.0 exomes release HT itself. I vote we resolve this thread